File: app/src/main/java/com/innoveworkshop/gametest/MainActivity.java
package com.innoveworkshop.gametest;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Color;
import android.os.Bundle;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;

import com.innoveworkshop.gametest.engine.Circle;
import com.innoveworkshop.gametest.engine.GameObject;
import com.innoveworkshop.gametest.engine.GameSurface;

public class MainActivity extends AppCompatActivity {
    protected GameSurface gameSurface;
    protected Button upButton;
    protected Button downButton;
    protected Button leftButton;
    protected Button rightButton;

    protected Game game;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        gameSurface = (GameSurface) findViewById(R.id.gameSurface);
        game = new Game();
        gameSurface.setRootGameObject(game);

        setupControls();
    }

    private void setupControls() {
        upButton = (Button) findViewById(R.id.up_button);
        upButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                game.circle.position.y -= 10;
            }
        });

        downButton = (Button) findViewById(R.id.down_button);
        downButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                game.circle.position.y += 10;
            }
        });

        leftButton = (Button) findViewById(R.id.left_button);
        leftButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                game.circle.position.x -= 10;
            }
        });

        rightButton = (Button) findViewById(R.id.right_button);
        rightButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                game.circle.position.x += 10;
            }
        });
    }

    class Game extends GameObject {
        public Circle circle;

        @Override
        public void onStart(GameSurface surface) {
            super.onStart(surface);

            circle = new Circle(surface.getWidth() / 2, surface.getHeight() / 2, 100, Color.RED);
            surface.addGameObject(circle);
        }

        @Override
        public void onFixedUpdate() {
            super.onFixedUpdate();

            //circle.setPosition(circle.position.x + 1, circle.position.y + 1);
        }
    }
}

File: app/src/main/java/com/innoveworkshop/gametest/MainActivity.kt
package com.innoveworkshop.gametest

import android.graphics.Color
import android.os.Bundle
import android.view.MotionEvent
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity
import com.innoveworkshop.gametest.assets.Football
import com.innoveworkshop.gametest.engine.Circle
import com.innoveworkshop.gametest.engine.GameObject
import com.innoveworkshop.gametest.engine.GameSurface
import com.innoveworkshop.gametest.engine.Rectangle
import com.innoveworkshop.gametest.engine.Vector

class MainActivity : AppCompatActivity() {
    protected var gameSurface: GameSurface? = null
    protected var upButton: Button? = null

    protected var game: Game? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        gameSurface = findViewById<View>(R.id.gameSurface) as GameSurface
        game = Game()
        gameSurface!!.setRootGameObject(game)

        setupControls()
    }

    private fun setupControls() {
        upButton = findViewById<View>(R.id.up_button) as Button
        upButton!!.setOnClickListener { game!!.football!!.position.y = -100f; game!!.football!!.position.x = gameSurface!!.width.toFloat() / 2f}
    }

    override fun onTouchEvent(event: MotionEvent?): Boolean {
        event?.let {
            // Pass the touch event to the football instance
            game?.football?.onTouchEvent(it)
        }
        return super.onTouchEvent(event)
    }

    inner class Game : GameObject() {
        var football: Football? = null

        override fun onStart(surface: GameSurface?) {
            super.onStart(surface)

            football = Football(
                    surface!!.width / 2.toFloat(), 0f, 250f, Color.BLUE, 1f
            )

            surface.addGameObject(football!!)

        }

        override fun onFixedUpdate() {
            super.onFixedUpdate()
        }
    }
}